src/routes/api/urls.go

```diff
@@ -21,24 +22,67 @@ func UrlsRoute() func(router fiber.Router) {
 	urlsController = controllers.CreateUrlsController()
 
 	return func(router fiber.Router) {
-		router.Get("/", getAllUrls)
+		router.Get("/", security.OptionalJwtAuthMiddleware, getAllUrls)
 		router.Post("/", security.MandatoryJwtAuthMiddleware, createRandomUrl)
 		router.Put("/:shortcode", security.MandatoryJwtAuthMiddleware, createSpecificUrl)
 	}
 }
 
 // getAllUrls
 //
-//	@Summary		Get all urls
-//	@Description	Get all urls
+//	@Summary		Get all urls (for users or admins)
+//	@Description	Get all urls for the current user or all URLs for admins
 //	@Tags			urls
 //	@Accept			json
 //	@Produce		json
-//	@Success		200	{string}	string	"GetAllUsers"
+//	@Param			X-API-Key	header		string	false	"Admin API Key (optional)"
+//	@Param			userid		query		string	false	"Filter URLs by user ID (admin only)"
+//	@Success		200			{array}		dtos.UrlResponse
+//	@Failure		400			{object}	dtos.ErrorResponse	"Invalid user ID"
+//	@Failure		401			{object}	dtos.ErrorResponse	"Unauthorized"
+//	@Failure		500			{object}	dtos.ErrorResponse	"Failed to fetch URLs"
 //	@Router			/urls [get]
-//	@security		BearerToken
+//	@Security		BearerToken
+//	@Security		ApiKeyAuth
 func getAllUrls(ctx *fiber.Ctx) error {
-	return ctx.SendString("GetAllUsers")
+	if ctx.Get("Authorization") == "" || ctx.Get("X-API-Key") == "" {
+		return ctx.Status(fiber.StatusUnauthorized).JSON(dtos.CreateErrorResponse(fiber.StatusUnauthorized, "Missing JWT or API key"))
+	}
+	apiKey := ctx.Get("X-API-Key")
+	isAdmin := apiKey == config.AdminApiKey
+
+	var userId *uint64
+	var err error
+
+	if isAdmin {
+		userIdStr := ctx.Query("userid")
+		if userIdStr != "" {
+			parsedId, parseErr := strconv.ParseUint(userIdStr, 10, 64)
+			if parseErr != nil {
+				return ctx.Status(fiber.StatusBadRequest).JSON(dtos.CreateErrorResponse(fiber.StatusBadRequest, "Invalid user ID"))
+			}
+			userId = &parsedId
+		}
+	} else {
+		user, ok := ctx.Locals(config.LOCALS_USER).(*models.User)
+		if !ok {
+			return ctx.Status(fiber.StatusUnauthorized).JSON(dtos.CreateErrorResponse(fiber.StatusUnauthorized, "Unauthorized"))
+		}
+		userId = &user.ID
+	}
+
+	var urls []models.Url
```

Comment:
what is the difference in the if condition here ? 

and why do we want it to be different ? 
if the userId came from LOCALS or it came from query (in case of admin) how does it differ? the logic for fetching the URLs will be exactly same in both those cases.

Reply:
Oh, I'm sorry. As a beginner, I memorized it and didn't realize that the origin of the userId doesn't inherently change how we should fetch the URLs, so we can unify the fetching URL.

Reply:
Sir, top level JWT/X-API-KEY if() check is wrong. 
- But in isAdmin; here we can't check `userId` at LOCALS (and in query the `userId` will be string so we've to convert it) and vise-versa in users case(here user.ID is integer). So that's why it was a good approach to separate them using if-else:
- what we can add now is"
    - replace parsetId with userId
    - have a new if-else check after this completes which will check :
    - if userId != null then GetUrlsByUserId(userId)  //works for both admin and users
    else GetAllUrls(nil)  //only work for admin to get all urls.

- This will maintain the userId consistency and also it checks the null values.